src/Client/StatsD.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+namespace HelloFresh\Stats\Client;
+
+
+use HelloFresh\Stats\Client;
+use HelloFresh\Stats\HTTPMetricAlterCallback;
+use HelloFresh\Stats\Incrementer;
+use HelloFresh\Stats\State;
+use HelloFresh\Stats\Timer\StatsD as StatsDTimer;
+use League\StatsD\Client as StatsDClient;
+
+class StatsD extends Base implements Client
+{
+    /** @var string */
+    protected $httpRequestSection;
+    /** @var HTTPMetricAlterCallback */
+    protected $httpMetricAlterCallback;
+    /** @var StatsDClient */
+    protected $client;
+
+    /**
+     * StatsD constructor.
+     *
+     * @param string $dsn statsd connection dsn
+     */
+    public function __construct($dsn)
+    {
+        $url = parse_url($dsn);
```

Comment:
[`parse_url`](http://php.net/parse_url) can return `NULL` you probably want to check for this.